airflow/gcp/hooks/base.py

```diff
@@ -323,8 +341,63 @@ def provide_gcp_credential_file_as_context(self):
                     pass
                 yield conf_file
             finally:
-                if current_env_state is None:
-                    if CREDENTIALS in os.environ:
-                        del os.environ[CREDENTIALS]
-                else:
-                    os.environ[CREDENTIALS] = current_env_state
+                self.restore_env_variable(current_env_state)
+
+    @staticmethod
+    def build_gcp_conn(
+        key_file_path: Optional[str] = None,
+        scopes: Optional[Sequence[str]] = None,
+        project_id: Optional[str] = None,
```

Comment:
It's a hook method, so it should read a project_id from hook also.

Reply:
My idea was to keep this method stateless. Part of the reason was the fact it's used in `provide_gcp_credentials_from_service_key` which is also a static method.

Reply:
We discussed it offline